pkg/download/classic/download.go

```diff
@@ -281,32 +257,37 @@ func downloadAndUnmarshalConfig(client client.ConfigClient, theApi api.API, valu
 		return nil, err
 	}
 
-	values, found := data[theApi.PropertyNameOfGetAllResponse]
+	vals, found := data[theApi.PropertyNameOfGetAllResponse]
 	if !found {
 		return []map[string]any{data}, nil
 	}
 
 	var res []map[string]any
-	err = mapstructure.Decode(values, &res)
+	err = mapstructure.Decode(vals, &res)
 	if err != nil {
 		return []map[string]any{}, err
 	}
-	if theApi.ID == api.KeyUserActionsWeb { //clean unwanted configs
-		return filterResponses(res, value)
+
+	if theApi.CheckEqualFunc != nil {
+		res = slices.DeleteFunc(res, func(m map[string]any) bool {
+			var remove bool
+			for _, r := range res {
+				remove = !theApi.CheckEqualFunc(m, r)
+			}
+			return remove
+		})
 	}
-	return res, nil
-}
 
-func filterResponses(res []map[string]any, value value) ([]map[string]any, error) {
-	for _, v := range res {
-		if v["meIdentifier"] == value.value.Id {
-			return []map[string]any{v}, nil
-		}
+	if theApi.PropertyNameOfIdentifier != "" {
+		return slices.DeleteFunc(res, func(m map[string]interface{}) bool {
+			return m[theApi.PropertyNameOfIdentifier].(string) != value.value.Id
```

Comment:
we need a way to identify the matching object in the payload. To know its identifier (key) the `PropertyNameOfIdentifier` value is used